docs/examples/userguide/language_basics/casting_python.py

```diff
@@ -0,0 +1,18 @@
+from cython.cimports.cpython.ref import PyObject
+
+def main():
+
+    python_string = "foo"
+
+    ptr: p_void = cython.cast(cython.p_void, python_string)
+    adress_in_c: Py_intptr_t= cython.cast(Py_intptr_t, ptr)
+    address_from_void = adress_in_c        # address_from_void is a python int
+
+    ptr2: cython.pointer(PyObject) = cython.cast(cython.pointer(PyObject), python_string)
+    address_in_c2: Py_intptr_t= cython.cast(Py_intptr_t, ptr2)
+    address_from_PyObject = address_in_c2  # address_from_PyObject is a python int
```

Comment:
I think we can avoid cluttering the example with type declarations when the type is "obviously" safe to infer.
```suggestion
    # Note that the variables below are automatically inferred
    # as the correct pointer type that is assigned to them.
    # They do not need to be typed explicitly.
    
    ptr = cython.cast(cython.p_void, python_string)
    adress_in_c = cython.cast(Py_intptr_t, ptr)
    address_from_void = adress_in_c        # address_from_void is a python int

    ptr2 = cython.cast(cython.pointer(PyObject), python_string)
    address_in_c2 = cython.cast(Py_intptr_t, ptr2)
    address_from_PyObject = address_in_c2  # address_from_PyObject is a python int
```